Improve equilibration and `md` configuration, improve analysis step

David-Araripe opened the pull request.

David-Araripe committed:
apply shake to the hydrogens in the beginning to allow Q to solve suboptimal bond geometries from auto-added hydrogens

David-Araripe committed:
do not apply shake to the solvents on equilibration steps 2 and 3

David-Araripe committed:
set shake_solute to be always off

David-Araripe committed:
fix bug when passing a custom experimental key

David-Araripe committed:
improve comments and error messages

jesperswillem reviewed (commented):
Is there a reason to turn off shake solute? We need it for 2fs timesteps.

David-Araripe committed:
Input files now follow the order when setting SHAKE: solvent, hydrogens, solute.

Now, setting the FEP run to 2fs will apply shake to all hydrogens in the simulation. Solvent SHAKE is set to off in case solvent contains other solvents like POPC.

shake_hydrogens already handles shake settings for HOH molecules.

David-Araripe committed:
use pd.isna(); robust to None and np.NaN; revert prepare_df to staticmethod

David-Araripe committed:
shake solvent always set to on since we use rigid water model

David-Araripe committed:
increase the step interval for logging (output), maching non_bond

David-Araripe commented:
Hi @jesperswillem, we actually _don't_ want `shake_solute` in place for production FEP using either 1fs or 2fs timesteps. The reason for it is that this setting applies SHAKE to _all_ protein atoms, including, for example C-C, C-N and C-O bonds, which shouldn't be restrained in first place. You can see this setting is turned off on an example file:
<img width="1087" height="359" alt="Screenshot 2025-12-12 at 12 21 39" src="https://github.com/user-attachments/assets/9cb27bec-d783-4227-8f59-1fb1603caa18" />

With respect to the hydrogens, we're currently making use of the `shake_hydrogens` as well as `shake_solvent`. Considering we're typically using TIP3, a rigid water model, it makes sense to have `shake_solvent` always set to on, regardless of the timestep.

The option `shake_hydrogens` is left as "STEPTOGGLE" which will be set by QligFEP as "off" for 1fs simulations and "on" for 2fs simulations. Here's a screenshot of the description of the different shake configurations in Q's manual:
<img width="1022" height="169" alt="Screenshot 2025-12-12 at 12 14 56" src="https://github.com/user-attachments/assets/f466035e-ea0d-41bf-923f-4a2ca50d22aa" />

David-Araripe committed:
Add method to reindex PDB residues and do that prior to cystein bond definition

David-Araripe commented:
Hi @jesperswillem I'm merging this into this new `dev` branch which I'll use to create the tests
